src/components/componentUtils.tsx

```diff
@@ -33,30 +34,45 @@ export function allZero(data: PieData[]) {
   return allZero;
 }
 
-export function LineChartWrapper(type: string, id?: string, lineType?: string) {
+type Props = {
+  type: string;
+  id?: string;
+};
+
+export function LineChartWrapper({ type, id }: Props) {
   const [data, setData] = useState<ILineChartResponse | null>();
-  const { location } = useContext(LocationContext);
   const { setUser } = useContext(GlobalContext);
+  const { location } = useContext(LocationContext);
+  const [loading, setLoading] = useState(false);
 
   async function getData() {
+    setLoading(true);
     if (type === "detector") {
       const response: ApiResponse = await GetLinePlotData(id!);
       setData(response.Unwrap(setUser));
     } else if (type === "location") {
-      if (location && lineType) {
+      if (location) {
         const response: ApiResponse = await GetLinePlotDataByLocation(
           location.id,
-          lineType,
         );
-        setData(response.Unwrap(setUser));
+        const data = response.Unwrap(setUser);
+        //TODO: make all the three diagrams
```

Comment:
We should create an issue on stuff like this, dont let unnecessary comment inside the code or if this is work in progress, than name the PR like the Developer handbook says.

Reply:
cleared